packages/block-editor/src/hooks/duotone.js

```diff
@@ -186,15 +199,10 @@ function addDuotoneAttributes( settings ) {
  */
 const withDuotoneControls = createHigherOrderComponent(
 	( BlockEdit ) => ( props ) => {
-		const hasDuotoneSupport = hasBlockSupport(
-			props.name,
-			'color.__experimentalDuotone'
-		);
-
 		return (
 			<>
 				<BlockEdit { ...props } />
-				{ hasDuotoneSupport && <DuotonePanel { ...props } /> }
+				<DuotonePanel { ...props } />
```

Comment:
Why did you move this check inside? It seems more natural here and has the added benefit that React doesn't have to bother rendering or running any of the hooks when it's not supported.

Reply:
I moved it back to the original position.

I inlined them with the rest while I was working on this to gain a clear picture of the conditions for visibility. Yeah, it's more performant the way it's now. After being more involved with these and with the changes, I'm also confident that the code is readable.